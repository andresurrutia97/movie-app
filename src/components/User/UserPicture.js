import React from 'react';
import PropTypes from 'prop-types';

import defaultUser from '../../img/default-user-image.png';

const UserPicture = ({ img }) => {
  return (
    <img
      className="h-32 w-32 mb-5 m-auto rounded-full shadow-lg shadow-md md:m-0 md:mr-12"
      src={img}
      alt="profile"
    />
  );
};

UserPicture.propTypes = {
  img: PropTypes.string,
};

UserPicture.defaultProps = {
  img: defaultUser,
};

export default UserPicture;
