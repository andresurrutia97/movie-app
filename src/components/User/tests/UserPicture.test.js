import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import UserPicture from '../UserPicture';
import defaultImg from '../../../img/default-user-image.png';

describe('<UserPicture/>', () => {
  const component = <UserPicture />;
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(component);
  });

  it('should render correctly ', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should contain an image source equal to the default user image path', () => {
    expect(wrapper.props().src).toBe(defaultImg);
  });

  it('should contain an image source equal to the path given', () => {
    wrapper.setProps({ img: 'path.jpg' });
    expect(wrapper.props().src).toBe('path.jpg');
  });
});
