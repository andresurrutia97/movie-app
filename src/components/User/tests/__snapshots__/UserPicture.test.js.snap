// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserPicture/> should render correctly  1`] = `
<img
  alt="profile"
  className="h-32 w-32 mb-5 m-auto rounded-full shadow-lg shadow-md md:m-0 md:mr-12"
  src="default-user-image.png"
/>
`;
